### ISSUE TITLE
Multiple GPU training issue in Kaggle

### ISSUE THREAD

**Issue by 2019331099-Rabbi** [ultralytics/ultralytics#7843]
### Search before asking

- [x] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Multi-GPU

### Bug

Hello!!!
I have implemented KFold(3 folds) for my dataset. I have set two GPUs for training. The problem arises here. Look at the following code.

import time
os.environ['WANDB_DISABLED'] = 'true'
project = 'kfold_demo'

results = list()
for i in range(ksplit-2):
    model = YOLO('yolov8m.pt')
    dataset_yaml = yaml_paths[i]
    print(f"Training for fold={i} using {dataset_yaml}")
    model.train(data=dataset_yaml,
            batch=16,
            project=project,
            epochs=1,
            verbose=False,
            workers=28,
            device=[0, 1]
    )
    result = model.metrics # Metrics on validation set
    results.append(result) # save output metrics for further analysis
    

When I use a single GPU for training, the **results** update successfully. But when I set `device=[0, 1]` for multiple GPUs the **results** update with `None` values([None, None, None] for 3 folds). Why is that happening?

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@2019331099-Rabbi hello! Thanks for reaching out with your issue regarding multi-GPU training on Kaggle using YOLOv8.

It seems like you're encountering a problem where the `results` are not being updated correctly when training with multiple GPUs. This could be related to how the distributed training process aggregates and returns the metrics after each fold. When using multiple GPUs, the metrics might not be synchronized properly across the devices, leading to `None` values in your results.

To resolve this, ensure that you're using the latest version of YOLOv8, as updates may contain fixes for multi-GPU training issues. Additionally, consider checking the multi-GPU training documentation on our [Ultralytics Docs](https://docs.ultralytics.com) to verify that your setup aligns with the recommended practices.

If the issue persists, please open a new issue on the YOLOv8 GitHub repository with detailed information about your environment and a minimal reproducible example, if possible. This will help us investigate the problem more effectively.

Thank you for contributing to the YOLO community, and we appreciate your efforts in using YOLOv8! 🚀
